search/src/Components/wrapper/datasets.js

```diff
@@ -38,7 +52,9 @@ class DatasetComponent extends Component {
     render() {
         return (
             <React.Fragment>
-                {this.state.upload_id === "*" ? <DatasetList handleUploadid={this.handleUploadid} upload_list={this.state.upload_list}/> : <DatasetSummary upload_id={this.state.upload_id} handleUploadid={this.handleUploadid}/>}
+                {this.state.upload_id === "*"?
+                <DatasetList handleUploadid={this.handleUploadid} upload_list={this.state.upload_list} awaiting_list={this.state.awaiting_list} is_staff={this.state.is_staff}/>:
+                <DatasetSummary upload_id={this.state.upload_id} handleUploadid={this.handleUploadid}/>}
             </React.Fragment>
         )
```

Comment:
I think:
1. You need to make DatasetList only show data from a single API call, not concatenate two tables together.
2. You should be using if-else here instead of ?: for clarity:
```js
        if (this.state.upload_id === "*") {
            // Dataset listing
            if (!this.state.is_staff || this.state.awaiting_list.length == 0) {
                return (<DatasetList handleUploadid={this.handleUploadid} upload_list={this.state.upload_list} />);
            } else {
                return (
                    <React.Fragment>
                        <h2>Awaiting approval</h2>
                        <DatasetList handleUploadid={this.handleUploadid} upload_list={this.state.awaiting_list} awaiting=true />
                        <h2>Public</h2>
						<DatasetList handleUploadid={this.handleUploadid} upload_list={this.state.upload_list} />
                    </React.Fragment>
                );
            }
       } else {
           return (<DatasetSummary upload_id={this.state.upload_id} handleUploadid={this.handleUploadid}/>);
       }
```